pkg/envtest/crd.go

```diff
@@ -225,8 +225,18 @@ func CreateCRDs(config *rest.Config, crds []*apiextensionsv1beta1.CustomResource
 	// Create each CRD
 	for _, crd := range crds {
 		log.V(1).Info("installing CRD", "crd", crd.Name)
-		if _, err := cs.ApiextensionsV1beta1().CustomResourceDefinitions().Create(crd); err != nil {
-			return err
+		var err error
+		if _, err = cs.ApiextensionsV1beta1().CustomResourceDefinitions().Get(crd.Name, metav1.GetOptions{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				_, err = cs.ApiextensionsV1beta1().CustomResourceDefinitions().Create(crd)
+			}
+			if apierrors.IsAlreadyExists(err) {
+				log.V(1).Info("CRD already exists, updating", "crd", crd.Name)
+				_, err = cs.ApiextensionsV1beta1().CustomResourceDefinitions().Update(crd)
+			}
+			if err != nil {
+				return err
+			}
```

Comment:
You also want some logic outside the get check, otherwise if `err == nil` (it already exists), we'll never update.

In fact, this is probably better as a `Create` check here still.

Reply:
we also need to load the resource version to do the update -- apparently we can't do an unconditional update here.

Reply:
I really needed caffeine when working on this. I am very sorry for that. Clearly, a conflict would never be the result of a GET 🤦‍♂
Now, I didn't know we'd forcibly need to set the resourceVersion when updating so TIL. THANKS!